src/katgpucbf/fgpu/main.py

```diff
@@ -125,6 +127,8 @@ def parse_narrowband(value: str) -> NarrowbandOutputDict:
                 match key:
                     case _ if key in kws:
                         raise ValueError(f"--narrowband: {key} specified twice")
+                    case "name":
+                        kws[key] = data
```

Comment:
No, there won't be spaces here, so I don't think it's necessary to do that kind of validation.

If something goes terribly wrong rather just let it crash IMO.

Reply:
There is nothing stopping you including almost any character (including spaces) when invoking fgpu from the command line - about the only thing you couldn't include would be a comma, since that splits up the keys-value pairs. At the moment I think it won't really matter. Once the name starts getting included in katcp sensors and the like, I like the idea of adding a check on the name to ensure that the sensor names are valid.

The product controller restricts stream names though, so this check would just be to protect people starting fgpu manually.